internal/cli/config.go

```diff
@@ -0,0 +1,33 @@
+package cli
+
+import (
+	"os"
+	"strings"
+
+	"github.com/knadh/koanf/parsers/json"
+	"github.com/knadh/koanf/providers/env"
+	"github.com/knadh/koanf/providers/file"
+	"github.com/knadh/koanf/v2"
+)
+
+type config struct {
+	k *koanf.Koanf
+}
+
+func newConfig() *config {
+	k := koanf.New(".")
+	k.Load(file.Provider("config.json"), json.Parser())
+	k.Load(file.Provider(os.ExpandEnv("$HOME/.config/summaraizer/config.json")), json.Parser())
+	k.Load(file.Provider(os.ExpandEnv("$HOME/.summaraizer/config.json")), json.Parser())
+	k.Load(env.Provider("SUMMARAIZER_", ".", func(s string) string {
+		envWithoutPrefix := strings.TrimPrefix(s, "SUMMARAIZER_")
+		return strings.Replace(strings.ToLower(envWithoutPrefix), "_", ".", -1)
+	}), nil)
+	return &config{
+		k: k,
+	}
+}
+
+func (c *config) GetString(path string) string {
+	return c.k.String(path)
```

Comment:
Here is some hint on how to leverage Go types to create a well organized config. No need to use error prone string keys. Plus you get some basic type checking for free. You also get to define some defaults in a nice way. You will need a bit more work on below snippet but you get the idea. So for now, current use of koanf package is okay but later try improving like below.

```diff
+type Config struct {
+	Slack  SlackConfig
+	Ollama OllamaConfig
+}
+
+type SlackConfig struct {
+	Token secret.Text
+}
+
+type OllamaConfig struct {
+	URL    string
+	Model  string
+	Prompt string
+}
+
+fun Load() *Config {
+	// add some default values if you like
+	cfg := &Config{
+		Ollama: OllamaConfig{
+			Prompt: "some default prompt",
+		},
+	}
+
+	ko := koanf.New("")
+	ko.Unmarshal("path/to/config", cfg)
+
+	return cfg
+}

```

Reply:
Done here https://github.com/ioki-mobility/summaraizer/pull/48/commits/db4448212cd2811e630f565d1b85bc08175decbb
Thanks fort he tip!